### PR TITLE
🎉 Destination Oracle: added custom JDBC parameters support.

### PR DESCRIPTION
**Pull request by pcsoni89**
## What
Add support for Oracle Destinations to take in a JDBC URL Params input to specify any additional JDBC parameters for the connection.

This is a follow up to https://github.com/airbytehq/airbyte/pull/10421
This is part of https://github.com/airbytehq/airbyte/issues/9501

## How
AbstractJdbcDestination can already handle jdbc parameters via the jdbc_url_params field, but the OracleDestination spec does not specify that field. We should update the spec to support additional jdbc parameters for Oracle destination.

## Recommended reading order
1. AbstractJdbcDestination.java
2. OracleDestination.java

## 🚨 User Impact 🚨
For existing destination oracle connections, a new parameter i.e. JDBC URL Params  will be displayed.It is an optional field and accpets key=value data. There won't be any impact on existing operations.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

</details>

### CODE REVIEW

**Comment by alexandertsukanov** [comment]
@pcsoni89 looks like the PR title is not following convention, please take a look here:
https://docs.airbyte.com/contributing-to-airbyte/updating-documentation/#issues-and-pull-requests

**Comment by alexandertsukanov** [comment]
Also, please verify if you have done all checkboxes from the **Updating a connector** section.

File: airbyte-integrations/connectors/destination-oracle/src/main/java/io/airbyte/integrations/destination/oracle/OracleDestination.java [hunk 1]
```diff
@@ -44,6 +44,8 @@ public class OracleDestination extends AbstractJdbcDestination implements Destin
   public static final String ENCRYPTION_KEY = "encryption";
   public static final String ENCRYPTION_METHOD_KEY = "encryption_method";
 
+  public static final String JDBC_URL_PARAMS_KEY = "jdbc_url_params";
```
Comment (alexandertsukanov) [inline airbyte-integrations/connectors/destination-oracle/src/main/java/io/airbyte/integrations/destination/oracle/OracleDestination.java:47]: Do we need to have this constant as public? if yes, please describe why

File: airbyte-integrations/connectors/destination-oracle/src/main/java/io/airbyte/integrations/destination/oracle/OracleDestination.java [hunk 2]
```diff
@@ -103,6 +105,9 @@ public JsonNode toJdbcConfig(final JsonNode config) {
     if (config.has("password")) {
       configBuilder.put("password", config.get("password").asText());
     }
+    if (config.has(JDBC_URL_PARAMS_KEY)) {
```
Comment (alexandertsukanov) [inline airbyte-integrations/connectors/destination-oracle/src/main/java/io/airbyte/integrations/destination/oracle/OracleDestination.java:108]: Do we need check if JDBC_URL_PARAMS_KEY **isEmpty()** here?

**Review by alexandertsukanov — CHANGES_REQUESTED** [review]
Leave, comments. Please, take a look.

File: airbyte-integrations/connectors/destination-oracle/src/main/java/io/airbyte/integrations/destination/oracle/OracleDestination.java (hunk 1 above)
Comment (pcsoni89) [reply airbyte-integrations/connectors/destination-oracle/src/main/java/io/airbyte/integrations/destination/oracle/OracleDestination.java:47]: The Constant is declared as public and final b/c we are accessing this in other classes as well like test classes.Similar declaration is done in other connectors like MySQL.

File: airbyte-integrations/connectors/destination-oracle/src/main/java/io/airbyte/integrations/destination/oracle/OracleDestination.java (hunk 2 above)
Comment (pcsoni89) [reply airbyte-integrations/connectors/destination-oracle/src/main/java/io/airbyte/integrations/destination/oracle/OracleDestination.java:108]: Have seen empty string support for other connectors (MySQL/MSSQL), that is why not added this condition.

**Comment by alexandertsukanov** [comment]
@pcsoni89 looks good could bump the connector version and run the tests again?

**Comment by alafanechere** [comment]
@alexandertsukanov are we up for a merge?

**Review by alexandertsukanov — APPROVED** [review]
This one looks good.

**Comment by alexandertsukanov** [comment]
/publish connector=connectors/destination-oracle-strict-encrypt

**Review by alexandr-shegeda — APPROVED** [review]
LGTM

**Comment by marcosmarxm** [comment]
@pcsoni89 @alexandertsukanov @alexandr-shegeda can publish and merge this PR?

**Comment by alexandr-shegeda** [comment]
closing this PR as out of date, opened a new one instead https://github.com/airbytehq/airbyte/pull/13841
cc @marcosmarxm

**Pull request closed without merging**